src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/MultiSourcePackageMetadataProvider.cs

```diff
@@ -255,15 +257,32 @@ private async Task<IPackageSearchMetadata> GetPackageMetadataAsync(PackageIdenti
                 tasks.Add(_localRepository.GetPackageMetadataFromLocalSourceAsync(identity, cancellationToken));
             }
 
+            if (_globalLocalRepositories != null)
+            {
+                _globalLocalRepositories.ForEach(x =>
+                    tasks.Add(x.GetPackageMetadataFromLocalSourceAsync(identity, cancellationToken)));
+            }
+
             IEnumerable<IPackageSearchMetadata> completed = (await Task.WhenAll(tasks))
                 .Where(m => m != null);
 
-            IPackageSearchMetadata master = completed.FirstOrDefault(m => !string.IsNullOrEmpty(m.Summary))
-                ?? completed.FirstOrDefault()
-                ?? PackageSearchMetadataBuilder.FromIdentity(identity).Build();
+            PackageSearchMetadataBuilder metadataBuilder = completed.Where(m => !string.IsNullOrEmpty(m.Summary)).Select(FromMetadata).FirstOrDefault()
+                ?? completed.Select(FromMetadata).FirstOrDefault()
+                ?? FromIdentity(identity);
 
-            return master.WithVersions(
-                asyncValueFactory: () => MergeVersionsAsync(identity, completed));
+            var clonedResult = metadataBuilder
+                .WithVersions(AsyncLazy.New(() => MergeVersionsAsync(identity, completed)))
+                .Build() as ClonedPackageSearchMetadata;
+
+            foreach (var package in completed)
```

Comment:
Can you help me understand if there's a reason to do this in the provider rather than copy the Url inside the builder?